.github/workflows/update-gemfiles.yml

```diff
@@ -3,78 +3,62 @@ name: Update Gemfiles
-      - name: Check if this branch is attached to a Pull Request
-        uses: 8bitjonny/gh-get-current-pr@2215326c76d51bfa3f2af0a470f32677f6c0cae9 # v2.2.0
-        id: pr
-        with:
-          filterOutClosed: true # Don't trigger on commits with closed PRs, including merges into `master`.
-      - if: steps.pr.outputs.pr_found == 'true'
+      - name: Checkout code
         uses: actions/checkout@v4
-      # And also, only execute if files that can affect gemfiles are modified.
-      - if: steps.pr.outputs.pr_found == 'true'
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+
+      - name: Check for changes in Gemfiles and related files
         uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
         id: filter
         with:
-          base: ${{ github.ref_name }}
           filters: |
             gemfile:
-              # Files that declare the dependency tree
               - Gemfile
               - Appraisals
               - datadog.gemspec
-              # Files that control gemfile generation
               - tasks/appraisal.rake
               - .github/workflows/update-gemfiles.yml
-              # The gem version is present in all lock files
               - lib/datadog/version.rb
-              # In case the generated files were updated manually or in a merge commit
-              - appraisal/**
-              - gemfiles/**
-      - if: steps.pr.outputs.pr_found == 'true' && steps.filter.outputs.gemfile == 'true'
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: '3.2'
-          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
-      - if: steps.pr.outputs.pr_found == 'true' && steps.filter.outputs.gemfile == 'true'
-        name: Ensure gemfiles/*.gemfile.lock match gem definition
-        run: bundle exec rake appraisal:lock
-      - if: steps.pr.outputs.pr_found == 'true' && steps.filter.outputs.gemfile == 'true'
-        name: Add all supported platforms to gemfiles/*.gemfile.lock
-        run: bundle exec rake appraisal:platform
-      - if: steps.pr.outputs.pr_found == 'true' && steps.filter.outputs.gemfile == 'true'
-        name: Remove obsolete gemfiles/*
-        run: bundle exec rake appraisal:clean
-      - if: steps.pr.outputs.pr_found == 'true' && steps.filter.outputs.gemfile == 'true'
-        name: Commit gemfiles changes, if any, back to the branch
-        uses: stefanzweifel/git-auto-commit-action@3ea6ae190baf489ba007f7c92608f33ce20ef04a # v4.16.0
+              - 'appraisal/**'
+              - 'gemfiles/**'
+
+      - name: Run update gemfiles if files changed
+        if: steps.filter.outputs.gemfile == 'true'
+        run: ruby .github/scripts/version_updater.rb
+
+      - name: Commit gemfiles changes, if any, back to the branch
```

Comment:
This only creates the commit and pushes changes - need to use a different repo to create the PR